src/main/scala/org/virtuslab/yaml/internal/load/reader/Tokenizer.scala

```diff
@@ -260,41 +262,27 @@ private[yaml] class YamlReader(in: CharSequence) extends Reader {
         }
       case _ => List(scalar)
 
-  inline private def peek(n: Int = 0): Option[Char] = Try(in.charAt(offset + n)).toOption
-  private def peekNext(): Option[Char]              = peek(1)
-  private def peekN(n: Int): String                 = (0 until n).map(peek(_)).flatten.mkString("")
-  private def isNextWhitespace                      = peekNext().exists(_.isWhitespace)
-
-  private def skipCharacter(): Unit = offset += 1
-  private def revert(n: Int): Unit  = offset -= n
-  private def skipN(n: Int): Unit   = (1 to n).foreach(_ => skipCharacter())
-
-  private def skipComment(): Unit =
-    while !(peek() == Some('\n') || peek() == Some('\r')) do skipCharacter()
-
-  private def read(): Char =
-    offset += 1
-    in.charAt(offset - 1)
-
   def skipUntilNextToken(): Unit =
-    while (peek() == Some(' ')) {
+    while (in.peek() == Some(' ')) {
       indent += 1
-      skipCharacter()
+      in.skipCharacter()
     }
 
-    if peek() == Some('#') then skipComment()
+    if in.peek() == Some('#') then skipComment()
 
-    if (peek() == Some('\n') || peek() == Some('\r')) then {
-      skipCharacter()
+    if (in.peek() == Some('\n') || in.peek() == Some('\r')) then {
+      in.skipCharacter()
       indent = 0
       skipUntilNextToken()
     }
 
   def skipUntilNextIndent(indentBlock: Int): Unit =
     indent = 0
-    while (peek() == Some(' ') && indent < indentBlock) {
+    while (in.peek() == Some(' ') && indent < indentBlock) {
       indent += 1
-      skipCharacter()
+      in.skipCharacter()
     }
 
+  private def skipComment(): Unit =
+    while !(in.peek() == Some('\n') || in.peek() == Some('\r')) do in.skipCharacter()
```

Comment:
I think, that we should move this logic to new method `in.isNextNewLine`

Reply:
I agree. What's more `\r\n` are two characters and we are skipping only one :/
I'll leave it as is because it needs more refactoring than just `in.isNextNewLine`